### ISSUE TITLE
[RFC] Examples folder organisation

### ISSUE THREAD

**Issue by Crissman** [optuna/optuna#1238]
The hierarchy of the `/examples` folder regarding pruning examples isn't clear.  In the `/pruning` folder, the examples in that folder aside from `simple.py`  are actually integration examples, which this one is not. Outside of that folder, there are other examples that include pruning, such as `chainer_simple.py` and `fastai_simple.py`.

Ok to relabel the `/pruning` folder `/integrations` and move `simple.py` above that folder to `/examples` and rename as `pruning_simple.py`?

**Comment by hvy** [comment]
The example directory is somewhat of a mess and I definitely agree that it should be cleaned up. Your approach sounds reasonable separating integration examples, although some concepts such as pruning are orthogonal. I personally prefer having another intermediate layer, something along the following.

example/
|_ integration/
|_ features/
  |_ pruning
  |_ visualization
  |_ ...

In any case, I think others might have other opinions so let me label this as "needs-discussion". And it's not really a bug so let me remove that label.

**Comment by hvy** [comment]
@keisuke-umezawa might have some comments on this issue.

**Comment by hvy** [comment]
I'd also be great if we could follow-up this with running daily example jobs in a more fine grained manner than what we're doing now, running every example in a single step https://github.com/optuna/optuna/actions/runs/171512614/workflow. I'd make debugging somewhat easier.

**Comment by Crissman** [comment]
After some thought about this, I think it would be better for us to group the examples by framework, like `pytorch`, `xgboost`, instead of by `pruning` or `integration`.

@hvy Don't follow your comment on Github actions, so let's discuss offline.

**Comment by hvy** [comment]
> @hvy Don't follow your comment on Github actions, so let's discuss offline.

If you for instance look at https://github.com/optuna/optuna/runs/1801496728?check_suite_focus=true, you can see that all examples are executed under a single large step (taking more than 3 hours). I meant splitting this up into multiple steps.

![Screen Shot 2021-02-01 at 13 08 59](https://user-images.githubusercontent.com/5983694/106413977-d238bf00-648e-11eb-979e-043b89b94ac4.png)

**Comment by Crissman** [comment]
Gotcha. Ok to make a PR to group examples by platform, and update the github actions to test by directory?

**Comment by hvy** [comment]
Sounds good to start working on a PR (or PRs) 👍

**Comment by hvy** [comment]
Let me close this issue since it's resolved. By the way, examples have been moved to https://github.com/optuna/optuna-examples.
